operators/constellation-node-operator/internal/cloud/gcp/client/scalinggroup.go

```diff
@@ -103,8 +103,7 @@ func (c *Client) GetAutoscalingGroupName(scalingGroupID string) (string, error)
 }
 
 // ListScalingGroups retrieves a list of scaling groups for the cluster.
-func (c *Client) ListScalingGroups(ctx context.Context, uid string) ([]cspapi.ScalingGroup, error) {
-	results := []cspapi.ScalingGroup{}
+func (c *Client) ListScalingGroups(ctx context.Context, uid string) (results []cspapi.ScalingGroup, retErr error) {
```

Comment:
Since you explicitly return the error in all cases anyway and I think that named returns sometime are a footgun, I'd define the retErr in the function itself and explicitly return in all cases (as already done)